### PR TITLE
Fix #279 - bank search not working when using two words split by space

### LINKED ISSUE #279: Bank search not working when using two words split by space
Hi,

when using for example `new FpBanks(..).search(new BanksCriteria().withSearch("Sparkasse Heid"))`
I don't get any results, even though when using finapi `https://sandbox.finapi.io/api/v1/banks` directly for given search string i get 4 results.
The thing is that url builder replaces whitespace with %2B instead %20 which doesn't work with finapi.

### PR DESCRIPTION
**Pull request by bzagozda**

### CODE REVIEW

**Comment by proshin-roman** [comment]
@bzagozda thanks for your pull request - I really appreciate your contribution to the project! I will do the code review ASAP 👍

File: src/main/java/org/proshin/finapi/primitives/pair/SpaceSeparatedPair.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.proshin.finapi.primitives.pair;
+
+public final class SpaceSeparatedPair extends UrlEncodedPair {
```
Comment (proshin-roman) [inline src/main/java/org/proshin/finapi/primitives/pair/SpaceSeparatedPair.java]: Please, use "Composition over inheritance" principle: classes must be final and the only way to extend them - add wrappers.

File: src/main/java/org/proshin/finapi/primitives/pair/SpaceSeparatedPair.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    public SpaceSeparatedPair(String name, String value) {
```
Comment (proshin-roman) [inline src/main/java/org/proshin/finapi/primitives/pair/SpaceSeparatedPair.java]: please mark all parameters as `final`

File: src/main/java/org/proshin/finapi/primitives/pair/SpaceSeparatedPair.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        super(name, value);
+    }
+
+    @Override
+    public String getValue() {
+        return super.getValue().replaceAll("\\+", " ");
```
Comment (proshin-roman) [inline src/main/java/org/proshin/finapi/primitives/pair/SpaceSeparatedPair.java]: it seems that `+` is the right encoding for space symbols: https://stackoverflow.com/questions/1634271/url-encoding-the-space-character-or-20
I think the real issue is that the "plus" symbol is encoded one more time for some reason. I've checked the code quickly but couldn't identify the problem. Could you check it please?

**Comment by ptrepka** [comment]
Hello, I faced the same problem and I found a bug in the library. It double encode parameters, because UriBuilder encode query params by itself, so the problem is that paramaters are encoded before with UrlEncodedPair class. So to the uri builder should be passed "raw" params.

**Comment by proshin-roman** [comment]
Hi @ptrepka ! Thank you for your comment! This pull request is waiting for changes for quite a long time, and even initially I was against of such a simple hack. And by coincidence I had the same bug in another project where I was able to find the right solution. The problem is that `URLEncoder` that is currently used in the library is not supposed to be used for URL's query part encoding - it's only for form parameter. And there is another util class form Spring `org.springframework.web.util.UriUtils` that provides method `encodeQueryParam` that works exactly as it's required by RFC 3986. I'm going to make another pull request and fix the original bug ticket #279 .

**Comment by ptrepka** [comment]
@proshin-roman Thanks for the fast answer. I will be waiting for the results :)

**Comment by proshin-roman** [comment]
The issue is going to be fixed by PR #357

**Pull request closed without merging**